dask/dataframe/groupby.py

```diff
@@ -2119,10 +2122,11 @@ def nunique(self, split_every=None, split_out=1):
     @derived_from(pd.core.groupby.SeriesGroupBy)
     def aggregate(self, arg, split_every=None, split_out=1):
         result = super().aggregate(arg, split_every=split_every, split_out=split_out)
-        if self._slice:
+        if self._slice is not None:
             result = result[self._slice]
 
         if not isinstance(arg, (list, dict)) and isinstance(result, DataFrame):
+            assert len(result.columns) == 1
```

Comment:
It was intentional, but I'm not certain it's correct and was trying to reason through it - If we have a single aggregation I _think_ it should only have a single column, but if there _were_ multiple, why is column 0 special?

It did "mask" [my issue](https://github.com/dask/dask/pull/9020#discussion_r864436331) a bit - instead of returning a dataframe with all columns (because the `if self._slice` was skipped - which would also be wrong, but was expected at first glance), I would get the wrong column (because the column named `1` was the 0th column).